sdk/core/core-lro/test/lro.spec.ts

```diff
@@ -1631,719 +1836,573 @@ matrix([["createPoller", "LroEngine"]] as const, async function (implName: Imple
-              headers: createDoubleHeaders({
-                pollingPath,
-                headers: { "Retry-After": "0" },
-              }),
-            },
-            {
-              method: "GET",
-              path: pollingPath,
-              status: 200,
-            },
+        it("should handle putAsyncRelativeRetryInvalidHeader ", async () => {
+          await assertError(
+            runLro({
+              routes: [
+                {
+                  method: "PUT",
+                  status: 200,
+                  body: `{ "properties": { "provisioningState": "Creating"}, "id": "100", "name": "foo" }`,
+                  headers: createDoubleHeaders({
+                    pollingPath: "/foo",
+                    headers: { "Retry-After": "/bar" },
+                  }),
+                },
+              ],
+            }),
             {
-              method: "GET",
-              path: path,
-              status: 200,
-            },
-          ],
+              statusCode: 404,
+            }
+          );
         });
-        assert.equal(response.statusCode, 200);
-      });
 
-      it("should handle putAsyncRelativeRetryNoStatus ", async () => {
-        const path = "/error/putasync/retry/nostatus";
-        const pollingPath = `/error/putasync/retry/failed/operationResults/nostatus`;
-        const response = await runLro({
-          routes: [
-            {
-              method: "PUT",
-              path,
-              status: 200,
-              body: `{ "properties": { "provisioningState": "Creating"}, "id": "100", "name": "foo" }`,
-              headers: createDoubleHeaders({
-                pollingPath,
-                headers: { "Retry-After": "0" },
-              }),
-            },
+        it("should handle putAsyncRelativeRetryInvalidJsonPolling ", async () => {
+          const pollingPath = `/error/putasync/retry/failed/operationResults/invalidjsonpolling`;
+          await assertError(
+            runLro({
+              routes: [
+                {
+                  method: "PUT",
+                  status: 200,
+                  body: `{ "properties": { "provisioningState": "Creating"}, "id": "100", "name": "foo" }`,
+                  headers: createDoubleHeaders({
+                    pollingPath,
+                    headers: { "Retry-After": "0" },
+                  }),
+                },
+                {
+                  method: "GET",
+                  path: pollingPath,
+                  status: 200,
+                  body: `{ "status": "Accepted"`,
+                },
+              ],
+            }),
             {
-              method: "GET",
-              path,
-              status: 200,
-              body: `{ }`,
-            },
+              messagePattern: /Unexpected end of JSON input/,
+            }
+          );
+        });
+
+        it("should handle delete202RetryInvalidHeader ", async () => {
```

Comment:
Hi @deyaaeldeen I noticed this case behavior didn't match with [ours ](https://github.com/Azure/autorest.typescript/blob/8ce0b33072ab10380d48469ea0fcd88178f0d806/packages/autorest.typescript/test/rlcIntegration/lroRest.spec.ts#L1069-L1084) in RLC test. 

I guess this may be caused by underlying core will throw RestError by default and RLC will not throw any exception. If yes Could you use the wrapper function `assertDivergentBehavior` for all test cases in `LRO Sad scenarios` section.

So that these cases could describe the different behaviors both in HLC and RLC.

Reply:
`assertDivergentBehavior` is used only when the behavior between HLC and RLC is different and in this case it is not. I will change behavior for RLC in another PR and this test case will change.

Reply:
This has been addressed in https://github.com/Azure/azure-sdk-for-js/pull/23311